lib/common/cubit/login_cubit.dart

```diff
@@ -0,0 +1,20 @@
+import 'package:flutter_bloc/flutter_bloc.dart';
+import 'package:firebase_auth/firebase_auth.dart';
+import 'login_state.dart';
+
+class LoginCubit extends Cubit<LoginState> {
+  LoginCubit() : super(LoginInitial());
+
+  Future<void> login(String email, String password) async {
+    try {
+      await FirebaseAuth.instance.signInWithEmailAndPassword(
```

Comment:
- [ ] This FirebaseAuth instance should not be get like this. Instead, pass it through the constructor
- [ ] mark it as injectable 
- [ ] Use the `_auth.signInWithEmailAndPassword`
- [ ] use the `getIt<LoginCubit>()` to create it (with automatic constructor params filling)
```dart
@injectable
class LoginCubit extends Cubit<LoginState> {
  final FirebaseAuth _auth;

  LoginCubit(this._auth) : super(LoginInitial());

  Future<void> login(String email, String password) async {
    try {
```